types/utils.d.ts

```diff
@@ -18,6 +18,27 @@ declare global {
         : T extends ReadonlyMap<infer KeyType, infer ValueType>
         ? ReadonlyMap<DeepPartial<KeyType>, DeepPartial<ValueType>>
         : { [K in keyof T]?: DeepPartial<T[K]> };
+
+    type DeepRequired<Type> = Type extends Error
```

Comment:
```suggestion
    type DeepRequired<T> = T extends Error
```
@yauheni-deriv Just to be the same as others, Also I found [this](https://gist.github.com/esamattis/70e9c780e08937cb0b016e04a7422010) one which seems simpler, Would it work? 🤔 

By the way, Thanks for helping to add type utilities 🙏🏻

Reply:
thanks @farzin-deriv , will go with your suggestion, looks really simpler

Reply:
the only concern is about 'Function'
I deleted it from  type primitive